zigzag/workload/layer_attributes.py

```diff
@@ -222,6 +231,27 @@ def remove_invalid_layer_dims(self, layer_dim_sizes: LayerDimSizes, layer_name:
     def to_legacy_format(self):
         return self.data
 
+    def get_constraints(self) -> list[PermutationConstraint]:
+        static_posistions_dict: dict[int, LayerDim] = {}
+        static_posistions_and_sizes_dict: dict[int, tuple[LayerDim, int]] = {}
+        outer_loop = False
+        for count, (layer_dim, factor) in enumerate(self.data):
+            if (layer_dim.name == "*") and (factor is None):
```

Comment:
If a LayerDim instance with name = `*` is used in the code, I would create a constant in the `Constants` class to prevent hardcoding the `*` throughout. This makes it easier if we at some point want to rename the `"*"` notation to something else